### DISCUSSION TITLE
CORS Misconfiguration Access-Control-Allow-Origin

### DISCUSSION THREAD

**Discussion by Dharmesh-Makwana** [requarks/wiki#3056] [Help / Questions]
I got this on my wiki configuration (Azure App service - docker) . Please suggest to set allow origin.
I see in the header that Access-Control-Allow-Origin is *
---------------------------------------------------------------------------------------------------------
Description: Cross-origin resource sharing (CORS) is a browser mechanism that enables controlled access to resources located outside of a given domain. It extends and adds flexibility to the same-origin policy (SOP). However, it also provides the potential for cross-domain based attacks, if a website's CORS policy is poorly configured and implemented

Risk: 
The application reflects arbitrary origins in the Access-Control-Allow-Origin header, this means that absolutely any domain can access resources from the vulnerable domain.

**Comment by NGPixel** [comment]
Fixed by c57c9d901807c6cb88f7f4e443925332eccaea1f

**Reply by lidaof** [reply]
is there a way to add a whitelist? thanks!

**Comment by Dusty4848** [comment]
Hey @Dharmesh-Makwana @NGPixel, could I ask how it will be possible to add allowed Origins?
I'm using the GraphQL API from a different domain, which has stopped working due to the above fix.

Thanks

**Reply by NGPixel** [reply]
You would need to undo these commits in your own installation:

https://github.com/Requarks/wiki/commit/c57c9d901807c6cb88f7f4e443925332eccaea1f

v3.0 will have a proper way to manage CORS.

**Comment by papiroca-tm** [comment]
How can I now send requests to fetch wiki.js pages from a different domain?

what's the point in an API if it can't be accessed?

**Reply by NGPixel** [reply]
The GraphQL API is meant to be accessed from the same origin (wiki.js) or from an external script which is not bound to CORS restrictions. See my reply above on how to re-enable the CORS manually.

**Comment by rehanzia9** [comment]
@NGPixel I've committed the change you suggest but it still giving me CORS error

```Access to fetch at 'https://example-wiki.herokuapp.com/graphql' from origin 'http://localhost:3000' has been blocked by CORS policy: Response to preflight request doesn't pass access control check: No 'Access-Control-Allow-Origin' header is present on the requested resource. If an opaque response serves your needs, set the request's mode to 'no-cors' to fetch the resource with CORS disabled.```

how can I resolve this? I'm using the GraphQL API from a different domain

**Reply by SVidu96** [reply]
Undoing the given commit by @NGPixel resolved my CORS issue.
Note: I'm testing wiki.js in my localhost and my frontend also hosted in my localhost
